libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -21,6 +21,9 @@ export connection
 logScope:
   topics = "libp2p mplexchannel"
 
+when defined(lipp2p_network_protocols_metrics):
+  declareCounter libp2p_protocols_bytes, "total sent or received bytes", ["protocol", "direction"]
```

Comment:
btw, what happens for streams / protocols like quic that presumably don't use mplex?

Reply:
We'll need to add the increments in every `lpchannel` equivalent (and probably move the declaration of the counter to `Connection` directly)

Unfortunately, we currently don't have a central place where we could add this metric, because
1) readOnce & writeOnce can be called directly and do not call the parent method
2) and if they called the parent method, that would be wrong because each layer would add to the counter, where in reality we only want to increment in the muxer

Reply:
Keep in mind, that we already have some traffic metrics, but at the chronosstream layer - https://github.com/status-im/nim-libp2p/blob/unstable/libp2p/stream/chronosstream.nim#L31-L34, this was yet another one off attempt. At this rate we'll end up with a bunch of this everywhere, I think it's time we think of a more consistent approach?

Reply:
Yes, theses ones were made to be per agent ones. Maybe we could have some flags-defined labels?
So one could choose which flags is used during compilation: agent, protocol, direction, etc.

But if we activate all flags, we would have `nagent * nprotocol * ndirection` time value series, instead of `nagent + nprotocol + ndirection` one, so not ideal..

Reply:
Yeah, that's why I say that this is becoming unmanageable, we have this metrics spread across everywhere and they are now hard to track. I don't have a straightforward solution for this unfortunately and it does look like it's at a point where it requires some focused attention.